runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -452,16 +452,30 @@ class ROMClassCreationContext
 					return ClassNameMismatch;
 #undef J9WRONGNAME
 				}
-			} else if (shouldCheckPackageName() /* classname is null */
-					&& (classNameLength >= 5)
-					&& (0 == memcmp(className, "java/", 5))
-					&& !isClassAnon()
-			) {
-				/*
-				 * Non-bootstrap classloaders may not load nto the "java" package.
-				 * if classname is not null, the JCL or JNI has already checked it
-				 */
-				return IllegalPackageName;
+			} else { /* classname is null */
+				/* If a name was not provided with the load data, now is our first chance to check for a duplicate definition */
+				if(NULL != _javaVM && J9_VM_FUNCTION_VIA_JAVAVM(_javaVM, hashClassTableAt)(_classLoader, className, classNameLength) != NULL) {
+					PORT_ACCESS_FROM_PORT(_portLibrary);
+					U_8 *errorUTF = (U_8 *) j9mem_allocate_memory(classNameLength + 1, J9MEM_CATEGORY_CLASSES);
+					if (NULL != errorUTF) {
+						memcpy(errorUTF, className, classNameLength);
+						errorUTF[classNameLength] = '\0';
+					}
+					recordCFRError(errorUTF);
```

Comment:
Is it legal to pass NULL here?

Reply:
We do the same thing in the above case where the passed name and the name in the class file don't match (https://github.com/eclipse-openj9/openj9/blob/cddb8bf6b5369e216402500b677da356114a38f8/runtime/bcutil/ROMClassCreationContext.hpp#L435) , although where we catch the error probably needs to be adjusted to avoid calling strlen(NULL)